### DISCUSSION TITLE
The problem of data latency when network traffic is particularly high

### DISCUSSION THREAD

**Discussion by alleniverson33** [cisagov/Malcolm#517] [Troubleshooting] [answered]
When the network traffic is particularly high, the Suricata alarm shows a delay of one hour in the dashboard. Is there any good solution to this？
I feel like Logstash can't forward it anymore

**Comment by mmguero** [comment]
I don't understand what you're saying: what do you mean the suricata alarm? Just that the data is taking an hour to show up? Or does the data have incorrect time stamps? What are your system specs? Is it with or without a network sensor? Is the other data (zeek?) showing up correctly? 

I'm going to be on vacation until December 2nd, but I will follow up here when I return.

**Comment by alleniverson33** [comment]
> I don't understand what you're saying: what do you mean the suricata alarm? Just that the data is taking an hour to show up? Or does the data have incorrect time stamps? What are your system specs? Is it with or without a network sensor? Is the other data (zeek?) showing up correctly?
> 
> I'm going to be on vacation until December 2nd, but I will follow up here when I return.

malcolm k8s     
When I attack a target, theoretically Suricata can detect the attack record and display it on the dashboard
Testing in a testing environment is normal, as long as an attack occurs, the attack record can be displayed
But in production environments, network traffic may be particularly high, and attack records may not be real-time. It may take an hour to see the logs of the attack now
I'm not sure if it's a problem with Filebeat collection or if opensearch insertion is experiencing a bottleneck

**Comment by mmguero** [comment]
Converting to a troubleshooting discussion, we can continue the conversation there.

**Comment by mmguero** [comment, marked as answer]
Is Malcolm doing the traffic capture in an all-in-one/standalone mode, or are you using a network sensor running hedgehog linux?

What are the resources (CPU and memory, and are the disks HDD or SSD?) for the malcolm server and sensor, if applicable?

There are some things you can try to tweak resources for performance:

* [Suricata tuning](https://cisagov.github.io/Malcolm/docs/live-analysis.html#LiveAnalysisTuningSuricata)
* [environment variables](https://cisagov.github.io/Malcolm/docs/malcolm-config.html#MalcolmConfigEnvVars) in [logstash.env](https://github.com/cisagov/Malcolm/blob/main/config/logstash.env.example), particularly the `pipeline` variables and the memory allowances in `LS_JAVA_OPTS`. 

This is not an issue I've ever seen before.

**Comment by mmguero** [comment]
As I haven't heard from you here in 3 weeks, I'll assume this helped you go in the right direction. Best of luck!
